src/renderer/components/+preferences/select-metrics-dialog.tsx

```diff
@@ -0,0 +1,124 @@
+import "./select-metrics-dialog.scss";
+
+import React from "react";
+import { observer } from "mobx-react";
+import { observable } from "mobx";
+import { Dialog, DialogProps } from "../dialog/dialog";
+import { Checkbox } from "../checkbox/checkbox";
+import { Wizard, WizardStep } from "../wizard/wizard";
+import { userStore } from "../../../common/user-store";
+import { Select, SelectOption } from "../select/select";
+import { Icon } from "../icon/icon";
+import { components, PlaceholderProps } from "react-select";
+
+interface Props extends Partial<DialogProps> {
+}
+
+export enum ResourceType {
+  Cluster = "Cluster",
+  Node = "Node",
+  Pod = "Pod",
+  Deployment = "Deployment",
+  StatefulSet = "StatefulSet",
+  Container = "Container",
+  Ingress = "Ingress",
+  VolumeClaim = "VolumeClaim",
+  ReplicaSet = "ReplicaSet",
+  DaemonSet = "DaemonSet",
+}
+
+const Placeholder = observer((props: PlaceholderProps<any>) => {
+  const getPlaceholder = (): React.ReactNode => {
+    if (userStore.hiddenMetrics.size >= 1) {
+      return <>Metrics: {Array.from(userStore.hiddenMetrics).join(", ")}</>;
+    }
+    else {
+      return <>Select metrics...</>;
+    }
+  };
+
+  return (
+    <components.Placeholder {...props}>
+      {getPlaceholder()}
+    </components.Placeholder>
+  );
+});
+
+@observer
+export class SelectMetricsDialog extends React.Component<Props> {
+
+  @observable static isOpen = false;
+
+  static open() {
+    SelectMetricsDialog.isOpen = true;
+  }
+
+  static close() {
+    SelectMetricsDialog.isOpen = false;
+  }
+
+  close = () => {
+    SelectMetricsDialog.close();
+  };
+
+  onChange(values: ResourceType[]) {
+    values.map(value => {
+      if (userStore.hiddenMetrics.has(value)) {
+        userStore.hiddenMetrics.delete(value);
+      } else {
+        userStore.hiddenMetrics.add(value);
+      }
+    });
+  }
+
+  reset = () => {
+    userStore.hiddenMetrics.clear();
+    userStore.preferences.hideMetrics = false;
+  };
+
+  render() {
+    const { ...dialogProps } = this.props;
+    const header = <h5>Select metrics to hide</h5>;
+
+    return (
+      <Dialog
+        {...dialogProps}
+        isOpen={SelectMetricsDialog.isOpen}
+        close={this.close}
+      >
+        <Wizard header={header} done={this.close}>
+          <WizardStep contentClass="flow column" nextLabel={"Reset"} next={this.reset} prevLabel={"Close"}>
+            <div>
+              <Select
+                className={"SelectMetricsDialog"}
+                components={{ Placeholder }}
```

Comment:
Why are there double `{` here? Does that have special meaning, if not please remove.

Reply:
If I'm not mistaken first {} for JS expression, second {}  for { key: value }  JS object
Here is an example 
https://react-select.com/components#defining-components

Reply:
Yes thanks